sw/device/lib/dif/dif_edn.h

```diff
@@ -0,0 +1,384 @@
+ * - auto refresh: EDN sends reseed and generate commands to the associated
+ *   CSRNG instance. The API allows the user to set the CSRNG instantiate,
+ *   reseed and generate para meters, as well as the reseed frequency.
+ *
+ * Common set of operations for both boot-time and auto refresh modes:
+ *
+ *  - `dif_edn_init()`
+ *  - `dif_edn_configure()`
+ *
+ * Order of operations in boot-time request mode:
+ *
+ *  - `did_edn_boot_mode_start()`
+ *  - `dif_edn_stop()`
+ *
+ * Order of operations in auto refresh mode:
+ *
+ *  - `dif_edn_auto_mode_start()`
+ *  - `dif_edn_stop()`
+ *
+ * Remaining work:
+ *
+ * - Add error status interface.
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_edn_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifEdnToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifEdnToggleDisabled,
+} dif_edn_toggle_t;
+
+/**
+ * Hardware instantiation parameters for Entropy Distribution Network.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_edn_params {
+  /**
+   * The base address for the Entropy Distribution Network hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_edn_params_t;
+
+/**
+ * A handle to Entropy Distribution Network.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_edn {
+  dif_edn_params_t params;
+} dif_edn_t;
+
+/**
+ * The result of a Entropy Distribution Network operation.
+ */
+typedef enum dif_edn_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifEdnOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifEdnError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occurred.
+   */
+  kDifEdnBadArg = 2,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifEdnLocked = 3,
+  /**
+   * Indicates that the device is busy and cannot perform the requested
+   * operation.
+   */
+  kDifEndBusy = 4,
+} dif_edn_result_t;
+
+/**
+ * CSRNG additional input data parameters for instantiate and
+ * generate commands.
+ */
+typedef struct dif_edn_seed_material {
+  /**
+   * Number of uint32_t words in `data`. Up to 12 words can be
+   * set to initialize or reseed the CSRNG. CSRNG will extend the `data`
+   * to zeros if the provided value is less that 12.
+   */
+  uint32_t len;
+  /**
+   * Seed material used in CSRNG instantiate or generate call.
+   */
+  uint32_t data[12];
```

Comment:
Where does `12` come from?

Reply:
Added a constant to define the meaning of this value.